process/block/baseProcess.go

```diff
@@ -1015,11 +1016,12 @@ func (bp *baseProcessor) getHeadersFromPools(
 	nonce uint64,
 ) ([]data.HeaderHandler, [][]byte) {
 
-	headers := make([]data.HeaderHandler, 0)
-	headersHashes := make([][]byte, 0)
+	keys := cacher.Keys()
+	headers := make([]data.HeaderHandler, 0, len(keys)+1)
+	headersHashes := make([][]byte, 0, len(keys)+1)
 
 	//TODO: This for could be deleted when the implementation of the new cache will be done
-	for _, headerHash := range cacher.Keys() {
+	for _, headerHash := range keys {
```

Comment:
if we preallocated the lists, than you could use index, and instead of append put inside the index value.